### PR TITLE
Improve modeling dynamic docs

### PR DESCRIPTION
**Pull request by DavisVaughan**
This PR creates the bare bones structure for @topepo to add commits to with the goal of improving the modeling documentation.

### CODE REVIEW

**Comment by topepo** [comment]
Any ideas about this error during travis?

```r
Building with: R CMD build 
* checking for file ‘./DESCRIPTION’ ... OK
* preparing ‘generics’:
* checking DESCRIPTION meta-information ... OK
* installing the package to process help pages
* building the PDF package manual
Hmm ... looks like a package
Converting Rd files to LaTeX Error : /tmp/RtmppKuGNI/Rbuild2b4829972b25/generics/man/augment.Rd:19: zero-length inputs cannot be mixed with those of non-zero length
```

**Comment by DavisVaughan** [comment]
Yea, it's fixed in #17. When that gets merged I'll rebase. Should be an easy one to rebase on. If you can still add docs and run devtools::document then you should be able to safely ignore it

**Comment by DavisVaughan** [comment]
@topepo this is ready for you to add your documentation commits to now

File: R/augment.R [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+#' @param data A [data.frame()] or [tibble::tibble()] containing the original 
+#'  data that was used to produce the object `x`. 
+#' @param ... Addition arguments to `augment` method.
+#' @return A [tibble::tibble()] with information about data points.
+#' @section Methods:
+#' \Sexpr[stage=render,results=Rd]{generics:::methods_rd("augment")}
+#' 
+#' @export
+#' 
+augment <- function(x, data, ...) {
```
[the remark this answers is not in the document]
Comment (topepo) [reply R/augment.R]: Do @alexpghayes  and/or @dgrtwo have opinions?

[the remark this answers is not in the document]
Comment (hadley) [reply R/augment.R]: I would strongly argue in favour of including it.

[the remark this answers is not in the document]
Comment (dgrtwo) [reply R/augment.R]: I used to disagree but I've come around to agree with Hadley on this

**Pull request merged**